## Projecto de analisis de los autos mas vendidos en Ebay 

In [1]:
import pandas as pd
autos = pd.read_csv("autos.csv",encoding = "Latin-1")

## A continuación se muestran las caracteristicas y datos de las columnas

Tenemos 4 datos relevantes
1. #: indica el numero al que correspone la columna correspondiente
2. Column: indica el nombre de la columna 
3. Non-Null Count: Indica el conteo de datos no nulos almacenados en cada columna
4. Dtype: Me muestra el tipo de Datos

In [2]:
autos.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50000 entries, 0 to 49999
Data columns (total 20 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   dateCrawled          50000 non-null  object
 1   name                 50000 non-null  object
 2   seller               50000 non-null  object
 3   offerType            50000 non-null  object
 4   price                50000 non-null  object
 5   abtest               50000 non-null  object
 6   vehicleType          44905 non-null  object
 7   yearOfRegistration   50000 non-null  int64 
 8   gearbox              47320 non-null  object
 9   powerPS              50000 non-null  int64 
 10  model                47242 non-null  object
 11  odometer             50000 non-null  object
 12  monthOfRegistration  50000 non-null  int64 
 13  fuelType             45518 non-null  object
 14  brand                50000 non-null  object
 15  notRepairedDamage    40171 non-null  object
 16  date

## A coninuación se muestran las primeras 5 fialas del DataFrame. 

In [3]:
autos.head()

dateCrawled                                               name  \
0  2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1  2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2  2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3  2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4  2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   

   seller offerType   price   abtest vehicleType  yearOfRegistration  \
0  privat   Angebot  $5,000  control         bus                2004   
1  privat   Angebot  $8,500  control   limousine                1997   
2  privat   Angebot  $8,990     test   limousine                2009   
3  privat   Angebot  $4,350  control  kleinwagen                2007   
4  privat   Angebot  $1,350     test       kombi                2003   

     gearbox  powerPS   model   odometer  monthOfRegistration fuelType  \
0    manuell      158  andere  150,000km                    3      lpg   
1  automatik      286     7er  150,000km                    6   benzin   
2    manuell      102    golf   70,000km                    7   benzin   
3  automatik       71  fortwo   70,000km                    6   benzin   
4    manuell        0   focus  150,000km                    7   benzin   

        brand notRepairedDamage          dateCreated  nrOfPictures  \
0     peugeot              nein  2016-03-26 00:00:00             0   
1         bmw              nein  2016-04-04 00:00:00             0   
2  volkswagen              nein  2016-03-26 00:00:00             0   
3       smart              nein  2016-03-12 00:00:00             0   
4        ford              nein  2016-04-01 00:00:00             0   

   postalCode             lastSeen  
0       79588  2016-04-06 06:45:54  
1       71034  2016-04-06 14:45:08  
2       35394  2016-04-06 20:15:37  
3       33729  2016-03-15 03:16:28  
4       39218  2016-04-01 14:38:50

## A continuación se muestran los nombres de las columnas. 

Los nombres de las columnas usan el formato CamelCase. 
Este formato para la limpieza de datos puede dificultar el proceso de limpieza. 

Por eso, cambiaremos el formato de CamelCase a snakecase. 
Se mantendra la misma escencia de los nombre para no alterar el conjunto de datos. 



In [4]:
autos.columns

Index(['dateCrawled', 'name', 'seller', 'offerType', 'price', 'abtest',
       'vehicleType', 'yearOfRegistration', 'gearbox', 'powerPS', 'model',
       'odometer', 'monthOfRegistration', 'fuelType', 'brand',
       'notRepairedDamage', 'dateCreated', 'nrOfPictures', 'postalCode',
       'lastSeen'],
      dtype='object')

In [5]:
map_snakecase = {'dateCrawled': "date_crawled", 'name':"name", 'seller':"seller",'offerType':"offer_type",'price':"price",'abtest':"ab_test",
    'vehicleType': "vehicle_type", 'yearOfRegistration': "registration_year",'gearbox':"gear_box",'powerPS':"power_ps", 'model': "model",
    'odometer':"odometer", 'monthOfRegistration':"registration_month", 'fuelType':"fuel_type",'brand':"brand",
    'notRepairedDamage': "unrepaired_damage", 'dateCreated':"ad_created", 'nrOfPictures':"nr_of_pictures",'postalCode':"postal_code",
    'lastSeen':"last_seen"}

In [6]:
autos.columns = autos.columns.map(map_snakecase)
autos.head()

date_crawled                                               name  \
0  2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1  2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2  2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3  2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4  2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   

   seller offer_type   price  ab_test vehicle_type  registration_year  \
0  privat    Angebot  $5,000  control          bus               2004   
1  privat    Angebot  $8,500  control    limousine               1997   
2  privat    Angebot  $8,990     test    limousine               2009   
3  privat    Angebot  $4,350  control   kleinwagen               2007   
4  privat    Angebot  $1,350     test        kombi               2003   

    gear_box  power_ps   model   odometer  registration_month fuel_type  \
0    manuell       158  andere  150,000km                   3       lpg   
1  automatik       286     7er  150,000km                   6    benzin   
2    manuell       102    golf   70,000km                   7    benzin   
3  automatik        71  fortwo   70,000km                   6    benzin   
4    manuell         0   focus  150,000km                   7    benzin   

        brand unrepaired_damage           ad_created  nr_of_pictures  \
0     peugeot              nein  2016-03-26 00:00:00               0   
1         bmw              nein  2016-04-04 00:00:00               0   
2  volkswagen              nein  2016-03-26 00:00:00               0   
3       smart              nein  2016-03-12 00:00:00               0   
4        ford              nein  2016-04-01 00:00:00               0   

   postal_code            last_seen  
0        79588  2016-04-06 06:45:54  
1        71034  2016-04-06 14:45:08  
2        35394  2016-04-06 20:15:37  
3        33729  2016-03-15 03:16:28  
4        39218  2016-04-01 14:38:50

## A continuación se muestran las descripción de las columnas. 

- Columnas que contengan datos numericos expresados como cadenas
- Se busca las columnas donde los datos sean constantes, lo cual pueden ser descartados porque carecen de importancia

In [7]:
autos.describe(include = "all")

date_crawled         name  seller offer_type  price ab_test  \
count                 50000        50000   50000      50000  50000   50000   
unique                48213        38754       2          2   2357       2   
top     2016-03-30 19:48:02  Ford_Fiesta  privat    Angebot     $0    test   
freq                      3           78   49999      49999   1421   25756   
mean                    NaN          NaN     NaN        NaN    NaN     NaN   
std                     NaN          NaN     NaN        NaN    NaN     NaN   
min                     NaN          NaN     NaN        NaN    NaN     NaN   
25%                     NaN          NaN     NaN        NaN    NaN     NaN   
50%                     NaN          NaN     NaN        NaN    NaN     NaN   
75%                     NaN          NaN     NaN        NaN    NaN     NaN   
max                     NaN          NaN     NaN        NaN    NaN     NaN   

       vehicle_type  registration_year gear_box      power_ps  model  \
count         44905       50000.000000    47320  50000.000000  47242   
unique            8                NaN        2           NaN    245   
top       limousine                NaN  manuell           NaN   golf   
freq          12859                NaN    36993           NaN   4024   
mean            NaN        2005.073280      NaN    116.355920    NaN   
std             NaN         105.712813      NaN    209.216627    NaN   
min             NaN        1000.000000      NaN      0.000000    NaN   
25%             NaN        1999.000000      NaN     70.000000    NaN   
50%             NaN        2003.000000      NaN    105.000000    NaN   
75%             NaN        2008.000000      NaN    150.000000    NaN   
max             NaN        9999.000000      NaN  17700.000000    NaN   

         odometer  registration_month fuel_type       brand unrepaired_damage  \
count       50000        50000.000000     45518       50000             40171   
unique         13                 NaN         7          40                 2   
top     150,000km                 NaN    benzin  volkswagen              nein   
freq        32424                 NaN     30107       10687             35232   
mean          NaN            5.723360       NaN         NaN               NaN   
std           NaN            3.711984       NaN         NaN               NaN   
min           NaN            0.000000       NaN         NaN               NaN   
25%           NaN            3.000000       NaN         NaN               NaN   
50%           NaN            6.000000       NaN         NaN               NaN   
75%           NaN            9.000000       NaN         NaN               NaN   
max           NaN           12.000000       NaN         NaN               NaN   

                 ad_created  nr_of_pictures   postal_code            last_seen  
count                 50000         50000.0  50000.000000                50000  
unique                   76             NaN           NaN                39481  
top     2016-04-03 00:00:00             NaN           NaN  2016-04-07 06:17:27  
freq                   1946             NaN           NaN                    8  
mean                    NaN             0.0  50813.627300                  NaN  
std                     NaN             0.0  25779.747957                  NaN  
min                     NaN             0.0   1067.000000                  NaN  
25%                     NaN             0.0  30451.000000                  NaN  
50%                     NaN             0.0  49577.000000                  NaN  
75%                     NaN             0.0  71540.000000                  NaN  
max                     NaN             0.0  99998.000000                  NaN

## Pasemos a eliminar las columnas que contienen datos constantes. 

Las columnas de "seller" y "offer_type" contienen solo 2 valores unicos, por lo cual vamos a eliminar ambas columnas.

In [8]:
autos.drop(['seller','offer_type'],axis=1,inplace=True)

In [9]:
autos.columns

Index(['date_crawled', 'name', 'price', 'ab_test', 'vehicle_type',
       'registration_year', 'gear_box', 'power_ps', 'model', 'odometer',
       'registration_month', 'fuel_type', 'brand', 'unrepaired_damage',
       'ad_created', 'nr_of_pictures', 'postal_code', 'last_seen'],
      dtype='object')

## Comencemos transformando las columnas que contienen datos numericos expresados en cadenas. 

- Price
- odometer 

In [10]:
autos["price"].describe()

count     50000
unique     2357
top          $0
freq       1421
Name: price, dtype: object

In [11]:
autos["odometer"].describe()

count         50000
unique           13
top       150,000km
freq          32424
Name: odometer, dtype: object

### Pasemos a remplazar "Price": 

- El Caracter $ para obtener solo una cadena que contiene un valor numerico
- Trasformar de Str (cadena) a valor Flotante 

In [12]:
autos["price"]= autos["price"].str.replace("$","").str.replace(",","").astype(float)

### Pasemos a remplazar "odometer": 

- los caracteres "km" para obtener solo una cadena que contiene un valor numerico, al igual que el caracter de ","
- Trasformar de Str (cadena) a valor Flotante 
- Cambiar el nombre de la columna de odometer a odometer_km para que quede claro que los valores estan descritos en kms. 

In [13]:
autos["odometer"]= autos["odometer"].str.replace("km","").str.replace(",","").astype(float)

In [14]:
autos.rename(columns = {"odometer":"odometer_km"}, inplace = True)

In [15]:
autos.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50000 entries, 0 to 49999
Data columns (total 18 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   date_crawled        50000 non-null  object 
 1   name                50000 non-null  object 
 2   price               50000 non-null  float64
 3   ab_test             50000 non-null  object 
 4   vehicle_type        44905 non-null  object 
 5   registration_year   50000 non-null  int64  
 6   gear_box            47320 non-null  object 
 7   power_ps            50000 non-null  int64  
 8   model               47242 non-null  object 
 9   odometer_km         50000 non-null  float64
 10  registration_month  50000 non-null  int64  
 11  fuel_type           45518 non-null  object 
 12  brand               50000 non-null  object 
 13  unrepaired_damage   40171 non-null  object 
 14  ad_created          50000 non-null  object 
 15  nr_of_pictures      50000 non-null  int64  
 16  post

### Ahora que tenemos los datos de "price" y "odometer" en datos numericos

Pasemos a verificar si tenemos valores atipicos en ambas columnas

Analicemos: 
1. Cuantos valores unicos tenemos en cada columna
2. Veamos la descripción de cada columna para ver si los valores max/min/mediana/media para ver si hay valores atipicos
3. Veamos el conteo de valores para conocer los valores que atipicos con exactitud

### Veamos la columna de price:

In [16]:
autos["price"].unique().shape[0]

2357

In [17]:
autos["price"].describe()

count    5.000000e+04
mean     9.840044e+03
std      4.811044e+05
min      0.000000e+00
25%      1.100000e+03
50%      2.950000e+03
75%      7.200000e+03
max      1.000000e+08
Name: price, dtype: float64

### Segun nuestro analisis anterior el 75% de nuestros datos se encuentra en la siguiente franja de prcios. 

- 25%: 1.100 
- 50%: 29.500
- 75%: 72.000

### Podriamos determinar que el rango de precios esta entre los:

### 1000 - 72.000

Vemos un poco mas los datos a fondo para determinar valores atipicos y confirmar que el rango seleccionado es correcto

In [18]:
autos["price"].value_counts().sort_index(ascending = False).head(5)

99999999.0    1
27322222.0    1
12345678.0    3
11111111.0    2
10000000.0    1
Name: price, dtype: int64

In [19]:
autos["price"].value_counts().sort_index(ascending = True).head(5)

0.0    1421
1.0     156
2.0       3
3.0       1
5.0       2
Name: price, dtype: int64

## Notemos que tenemos valores atipicos, precios muy bajos y precios muy altos. 

Vamos a proceder a hacer la limpieza en de la columna "price"

Vamos a considerar solos las filas donde precio esta en un rango de 500-72000 Para evitar una perdida de datos muy alta

In [20]:
autos = autos[autos["price"].between(500.0,32000.0)]

In [21]:
autos["price"].describe()

count    44424.000000
mean      5620.952233
std       5871.615129
min        500.000000
25%       1500.000000
50%       3400.000000
75%       7600.000000
max      32000.000000
Name: price, dtype: float64

### Veamos la columna de odometer_km:

In [22]:
autos["odometer_km"].unique().shape[0]

13

In [23]:
autos["odometer_km"].describe()

count     44424.000000
mean     126265.757248
std       38758.790012
min        5000.000000
25%      125000.000000
50%      150000.000000
75%      150000.000000
max      150000.000000
Name: odometer_km, dtype: float64

### Tenemos rangos muy amplios en los el kilometraje de los autos en venta

Segun la descripción de la columna "odometer"

El 75% de los datos se encuentran entre los 125.000 y los 150.000

Pero exploremos un poco mas los datos a fondo. 

El objetivo es determinar un rango razonable, de esta manera delimitaremos y descartaremos algunos valores para facilitar la legibilidad del analisis

In [24]:
autos["odometer_km"].value_counts().sort_index(ascending = False)

150000.0    28633
125000.0     4798
100000.0     1993
90000.0      1642
80000.0      1348
70000.0      1143
60000.0      1077
50000.0       935
40000.0       741
30000.0       699
20000.0       637
10000.0       198
5000.0        580
Name: odometer_km, dtype: int64

In [25]:
autos["odometer_km"].value_counts().sort_index(ascending = True)

5000.0        580
10000.0       198
20000.0       637
30000.0       699
40000.0       741
50000.0       935
60000.0      1077
70000.0      1143
80000.0      1348
90000.0      1642
100000.0     1993
125000.0     4798
150000.0    28633
Name: odometer_km, dtype: int64

### Vamos a determinar un rango entre los 50.000 - 150.000

De esta manera delimitaremos la columna de Odometer

In [26]:
autos = autos[autos["odometer_km"].between(50000.0,150000.0)]

In [27]:
autos["odometer_km"].describe()

count     41569.000000
mean     133296.446871
std       28720.651371
min       50000.000000
25%      125000.000000
50%      150000.000000
75%      150000.000000
max      150000.000000
Name: odometer_km, dtype: float64

### Ahora que hemos limpiadpo las columnas "price" y "odometer_km"

Veamos como se nos quedo nuestros conjunto de datos. 

Nuestro conjunto de datos original tenia 50.000 filas y 20 columnas

Despues de nuestra limpieza, nuestro conjunto de datos tiene 41569 y 18 columnas. 

In [28]:
autos.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 41569 entries, 0 to 49999
Data columns (total 18 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   date_crawled        41569 non-null  object 
 1   name                41569 non-null  object 
 2   price               41569 non-null  float64
 3   ab_test             41569 non-null  object 
 4   vehicle_type        38153 non-null  object 
 5   registration_year   41569 non-null  int64  
 6   gear_box            39951 non-null  object 
 7   power_ps            41569 non-null  int64  
 8   model               39748 non-null  object 
 9   odometer_km         41569 non-null  float64
 10  registration_month  41569 non-null  int64  
 11  fuel_type           38621 non-null  object 
 12  brand               41569 non-null  object 
 13  unrepaired_damage   34530 non-null  object 
 14  ad_created          41569 non-null  object 
 15  nr_of_pictures      41569 non-null  int64  
 16  post

### Vamos a ver como estan representados los datos en las columnas  date_crawled, last_seen, and ad_created

In [29]:
autos[["date_crawled","ad_created","last_seen"]].describe()

date_crawled           ad_created            last_seen
count                 41569                41569                41569
unique                40345                   73                33878
top     2016-03-05 16:57:05  2016-04-03 00:00:00  2016-04-07 06:17:27
freq                      3                 1621                    8

### Analicemos a profundidad la columna ad_created (fecha en que se publico el anuncio en ebay)

1. La fecha esta descrita en el formato AAAA-MM-DD H:M:S. Sin embargo es una costante que las horas,minutos y los segundos sean 00:00:00 Podemos deducir que ebay crea el registro en su web del anuncion a las 00:00:00  

2. Ademas tenemos 73 valores unicos, esto se debe a que ebay solo crea como maximo, un registro al dia de los anuncios que se publican en ebay. 

3. El 2016 fue el año donde mas datos se crearon en el sitio web. Es decir, cuantos mas autos se subieron a la venta en la plataforma de Ebay. 

Incluso, podemos deducir, por los datos mostrados, que el mes 04 del año 2016 fue cuando mas autos se pusieron en venta en la plataforma de Ebay            

In [30]:
autos["ad_created"].value_counts(normalize=True, dropna=False).sort_index()

2015-08-10 00:00:00    0.000024
2015-09-09 00:00:00    0.000024
2015-11-10 00:00:00    0.000024
2015-12-05 00:00:00    0.000024
2015-12-30 00:00:00    0.000024
                         ...   
2016-04-03 00:00:00    0.038995
2016-04-04 00:00:00    0.037167
2016-04-05 00:00:00    0.011884
2016-04-06 00:00:00    0.003320
2016-04-07 00:00:00    0.001251
Name: ad_created, Length: 73, dtype: float64

### Analicemos a profundidad la columna date_crawled (Primera vez en que se rastreo el anuncio)

1. La fecha esta descrita en el formato AAAA-MM-DD H:M:S. El registro es mucho mas especifico ya que toma en cuenta la fecha y hora del rastreador. 

2. Tenemos 40345 valores unicos, lo que tiene sentido tomando en cuenta que el registro esta basado en la fecha y hora del rastreador. 

3. Se mantiene la constante de que en el 2016 se generaron mas registros. 

In [31]:
autos["date_crawled"].value_counts(normalize=True, dropna=False).sort_index()

2016-03-05 14:06:30    0.000024
2016-03-05 14:06:40    0.000024
2016-03-05 14:07:08    0.000024
2016-03-05 14:07:21    0.000024
2016-03-05 14:07:26    0.000024
                         ...   
2016-04-07 14:30:09    0.000024
2016-04-07 14:30:26    0.000024
2016-04-07 14:36:44    0.000024
2016-04-07 14:36:55    0.000024
2016-04-07 14:36:56    0.000024
Name: date_crawled, Length: 40345, dtype: float64

### Analicemos a profundidad la columna last_seen

1. La fecha esta descrita en el formato AAAA-MM-DD H:M:S. 

2. Tenemos 33878 valores unico. Lo que nos dice que puede que tengamos valores perdidos si tomamos en consideración los valores unicos de la fecha de registro del restrador (date_crawled). 
3. Se mantiene la constante de que en el 2016 se mostraron mas autos por ultima vez. 

In [32]:
autos["last_seen"].value_counts(normalize=True, dropna=False).sort_index()

2016-03-05 14:45:46    0.000024
2016-03-05 14:46:02    0.000024
2016-03-05 14:49:34    0.000024
2016-03-05 15:16:47    0.000024
2016-03-05 15:28:10    0.000024
                         ...   
2016-04-07 14:58:44    0.000072
2016-04-07 14:58:45    0.000024
2016-04-07 14:58:46    0.000024
2016-04-07 14:58:48    0.000072
2016-04-07 14:58:50    0.000072
Name: last_seen, Length: 33878, dtype: float64

### Analicemos a profundidad la columna de registration_year

1. La mayoria de los coches tienen año de matricula entre 1999 y 2008
2. Nuestro dato esta reflado en float
3. Mas del 50% de los coches publicados tienen fecha de matricula entre 2004 y 2008. 
4. La lista toma en consideración los registros de los anuncios hasta el 2016. Por lo cual, cualquier coche con matricula mayor a 2016, en definitiva es un error, 

In [33]:
autos["registration_year"].describe()

count    41569.000000
mean      2003.441940
std          6.933982
min       1910.000000
25%       1999.000000
50%       2004.000000
75%       2008.000000
max       2019.000000
Name: registration_year, dtype: float64

In [34]:
autos["registration_year"].value_counts(normalize=True, dropna=False).sort_index()

1910    0.000024
1934    0.000024
1937    0.000048
1941    0.000024
1948    0.000024
          ...   
2015    0.000144
2016    0.020953
2017    0.029421
2018    0.010561
2019    0.000024
Name: registration_year, Length: 72, dtype: float64

### Pasemos a limpiar las filas donde el registro de los autos sea superior a 2016

Definamos un rango donde tomemos en consideración el minimo que es 1910 y el maximo que es 2016 que es el año donde se exporto la lista de autos publicados. 

Vamos a eliminar todos los coches con año de matricula superior a 2016

In [35]:
autos = autos[autos["registration_year"].between(1910,2016)]

In [36]:
autos["registration_year"].value_counts(normalize=True)

2005    0.069864
2000    0.066456
2004    0.065955
2003    0.065679
2006    0.065504
          ...   
1941    0.000025
1953    0.000025
1910    0.000025
1948    0.000025
1952    0.000025
Name: registration_year, Length: 69, dtype: float64

### Analicemos cuales son las marcas de coche que mas se publican en Ebay. 

1. Analizaremos la cantidad de anuncios de cada marca en concreto
2. Selecionaremos las 15 marcas mas populares dentro de los anuncios de autos en ebay 
3. Obtendremos el precio medio de las 15 marcas mas populares

In [37]:
top_8_brand = autos["brand"].value_counts(normalize=True,dropna=False).head(8).index

In [38]:
brand_mean_prices = {}

In [39]:
for brand in top_8_brand:
    precio_medio = autos.loc[autos["brand"] == brand, "price"].mean().round()
    brand_mean_prices[brand] = precio_medio

### Vamos a crear conjunto de marcas basandonos en el precio medio. 

1. Establecemos rango de precios en funcion de los resultados mostrados a continuación

- 6000 - 8000: bmw, mercedes_benz, audi
- 4000 - 5999: volkswagen
- 2000 - 3999: ford,renault,peugeot, opel

In [40]:
print(brand_mean_prices)

{'volkswagen': 5070.0, 'bmw': 7527.0, 'mercedes_benz': 7292.0, 'opel': 3048.0, 'audi': 7736.0, 'ford': 3519.0, 'renault': 2461.0, 'peugeot': 3110.0}


### Para nuestro analsiis tambien queremos saber cual es la media de kilometraje basandonos en las principales 8 marcas de coches

In [41]:
brand_mean_odometer = {}

In [42]:
for brand in top_8_brand:
    kilometraje_medio = autos.loc[autos["brand"] == brand, "odometer_km"].mean().round()
    brand_mean_odometer[brand] = kilometraje_medio

In [43]:
print(brand_mean_odometer)

{'volkswagen': 135519.0, 'bmw': 138397.0, 'mercedes_benz': 137609.0, 'opel': 134262.0, 'audi': 138146.0, 'ford': 131247.0, 'renault': 133110.0, 'peugeot': 130670.0}


### Para nuestros analsis, vamos a mostrar el precio medio y el kilometraje medio de las 8 marcas mas vendidas. 
1. Creamos primero un pandas.series de cada diccionario brand_mean_odometer y brand_mean_prices

2. Creamos un padas.DataFrame a partir bmp_series
3. luego agregamos una nueva columna en el pandas.dataframe a partir de bmo_series

In [44]:
bmp_series = pd.Series(brand_mean_prices)

In [45]:
bmo_series = pd.Series(brand_mean_odometer)

In [46]:
brand_km_price_mean= pd. DataFrame(bmp_series, columns =["mean_price"])

In [47]:
brand_km_price_mean["mean_odometer"] = bmo_series

In [48]:
brand_km_price_mean

mean_price  mean_odometer
volkswagen         5070.0       135519.0
bmw                7527.0       138397.0
mercedes_benz      7292.0       137609.0
opel               3048.0       134262.0
audi               7736.0       138146.0
ford               3519.0       131247.0
renault            2461.0       133110.0
peugeot            3110.0       130670.0

### Saquemos algunas conclusiones de nuestro dataframe

- El kilimetraje_medio en las principales 8 marcas es > 130.000 km
- El precio_medio en las principales 8 marcas

In [49]:
test2 = autos["name"].value_counts().head(10).index

In [50]:
dict_price = {}
dict_brand = {}
dict_odometer = {}
dict_name = {}
dict_unrepaired_damage = {}

In [51]:
for x in test2: 
    price_mean = autos.loc[autos["name"] == x, "price"].mean().round()
    brand_model = autos.loc[autos["name"] == x, "brand"].unique()
    odometer_mean = autos.loc[autos["name"] == x, "odometer_km"].mean().round()
    model_name = autos.loc[autos["name"] == x, "model"]
    unrepaired_damage_name = autos.loc[autos["name"] == x, "unrepaired_damage"].unique()
    dict_price[x] = price_mean
    dict_brand[x] = brand_model
    dict_odometer[x] = odometer_mean
    dict_name[x] = model_name
    dict_unrepaired_damage[x] = unrepaired_damage_name
    

In [52]:
odometer_series = pd.Series(dict_odometer)

In [53]:
price_series = pd.Series(dict_price)

In [54]:
brand_series = pd.Series(dict_brand)

In [55]:
unrepaired_damage_series = pd.Series(dict_unrepaired_damage)

In [56]:
autos_final = pd.DataFrame(odometer_series, columns = ["odometer_km_mean"])

In [57]:
autos_final["price_mean"] = price_series

### A continuación saquemos otras estadisticas

1. Se muestra a continuación cuales son los autos mas anunciados en la pagina de ebay
2. Se muestra el kilometraje medio de los coches mas anunciados
3. Se muestra el precio medio de los cohes mas anunciados

In [58]:
autos_final

odometer_km_mean  price_mean
BMW_316i                     145133.0      1940.0
Volkswagen_Golf_1.4          134667.0      2343.0
BMW_318i                     131736.0      4610.0
BMW_320i                     140070.0      4264.0
Volkswagen_Polo              133284.0      2080.0
Opel_Corsa                   133929.0      1439.0
Opel_Corsa_1.2_16V           131727.0      2539.0
Ford_Fiesta                  127091.0      1884.0
Volkswagen_Golf              134717.0      4110.0
BMW_116i                     112115.0      7603.0

### Otra estadistica interesante es ver la diferencia de precio entre los coches con daños y sin daños.

In [59]:
test3 = list(autos["unrepaired_damage"].unique())

In [60]:
dict_unrepaired_price = {}

In [61]:
for x in test3:
    unrepaired_price_mean = autos.loc[autos["unrepaired_damage"] == x, "price"].mean()
    dict_unrepaired_price[x]= unrepaired_price_mean

In [62]:
print(dict_unrepaired_price)

{'nein': 5943.304447394623, nan: nan, 'ja': 2623.773892773893}


In [63]:
unrepaired_series = pd.Series(dict_unrepaired_price)

In [64]:
unrepaired_dataframe = pd.DataFrame(unrepaired_series, columns = ["mean_price"])

### A continuación visualizaremos el precio medio de los coches con daños y sin daños

- nein: Coches sin daños
- ja: Coches con daños 

In [65]:
unrepaired_dataframe

mean_price
nein  5943.304447
NaN           NaN
ja    2623.773893